lib/CGameInfoCallback.cpp

```diff
@@ -564,9 +572,34 @@ EPlayerStatus::EStatus CGameInfoCallback::getPlayerStatus(PlayerColor player, bo
 	return ps->status;
 }
 
-std::string CGameInfoCallback::getTavernGossip(const CGObjectInstance * townOrTavern) const
+std::string CGameInfoCallback::getTavernRumor(const CGObjectInstance * townOrTavern) const
 {
-	return "GOSSIP TEST";
+	std::string text = "", extraText = "";
+	if(gs->rumor.type == RumorState::RUMOR_NONE) // (version < 755 backward compatability
```

Comment:
Just a thought - what about marking all places for backwards compatibiltiy in some consistent format? Or even static_assert'ing that this version is not less than minimal supported?

This would allow us to remove all compatibility code once save version is increased instead of accumulating it over releases.